vocode/streaming/agent/state_agent.py

```diff
@@ -576,7 +576,7 @@ async def handle_state(self, state_id_or_label: str):
             )
             return
         self.json_transcript.entries.append(
-            StateAgentTranscriptHandleState(state_id=state["id"])
+            StateAgentTranscriptHandleState(state_id=state["id"], generated_label=state.get("generated_label", state["id"]))
```

Comment:
Here this is not clear. Might make more sense to set the generated_label to optional in the type and default to None.

Reply:
Fair enough.